kroxylicious-sample/src/test/java/io/kroxylicious/sample/SampleFetchResponseFilterTest.java

```diff
@@ -0,0 +1,164 @@
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.common.message.ApiMessageType;
+import org.apache.kafka.common.message.FetchResponseData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.MemoryRecordsBuilder;
+import org.apache.kafka.common.record.RecordBatch;
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.common.utils.ByteBufferOutputStream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.stubbing.Answer;
+
+import io.kroxylicious.proxy.filter.FilterInvoker;
+import io.kroxylicious.proxy.filter.FilterInvokers;
+import io.kroxylicious.proxy.filter.KrpcFilterContext;
+import io.kroxylicious.sample.config.SampleFilterConfig;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class SampleFetchResponseFilterTest {
+
+    private static final short API_VERSION = 0; // this is arbitrary for our filter, so set to 0
+    private static final String PRE_TRANSFORM_VALUE = "this is what the value will be transformed from";
+    private static final String NO_TRANSFORM_VALUE = "this value will not be transformed";
+    private static final String POST_TRANSFORM_VALUE = "this is what the value will be transformed to";
+    private static final String CONFIG_FIND_VALUE = "from";
+    private static final String CONFIG_REPLACE_VALUE = "to";
+    private static final ApiMessageType API_MESSAGE_TYPE = ApiMessageType.FETCH;
+
+    @Mock
+    private KrpcFilterContext context;
+
+    @Captor
+    private ArgumentCaptor<ApiMessage> apiMessageCaptor = ArgumentCaptor.forClass(ApiMessage.class);
+
+    private FilterInvoker invoker;
+
+    @BeforeEach
+    public void beforeEach() {
+        buildContextMock();
+        SampleFilterConfig config = new SampleFilterConfig(CONFIG_FIND_VALUE, CONFIG_REPLACE_VALUE);
+        SampleFetchResponseFilter filter = new SampleFetchResponseFilter(config);
+        invoker = FilterInvokers.from(filter);
+    }
+
+    /**
+     * Unit Test: Checks that transformation is applied when response data contains configured value.
+     */
+    @Test
+    public void willTransformFetchResponseTest() {
+        var headerData = new ResponseHeaderData();
+        var responseData = buildFetchResponseData(PRE_TRANSFORM_VALUE);
+        invoker.onResponse(ApiKeys.forId(API_MESSAGE_TYPE.apiKey()), API_VERSION, headerData, responseData, context);
```

Comment:
Do we need to go via the `invoker`? I think we can just call `filter.onFectchResponse` directly